lib/JSON/Validator.pm

```diff
@@ -420,6 +420,20 @@ sub _resolve {
   return $schema;
 }
 
+sub _absolutise_location {
+  my ($location, $base) = @_;
+  my $location_as_url = Mojo::URL->new($location);
+  return $location_as_url if $location_as_url->is_abs;
+  # definitely relative now
+  my $ref = blessed $base
+    or die "[JSON::Validator] \$ref location base was non-object";
+  if ($ref eq 'Mojo::File') {
```

Comment:
Instead of using $ref, then just do `$base->isa('Mojo::File')` instead. And I don't want the `die(...)` above. This is not user input, so a unit test should be enough.